src/main/java/rcms/utilities/daqexpert/reasoning/logic/failures/CmsswCrashes.java

```diff
@@ -0,0 +1,94 @@
+    private List<Triple<Long, Integer, Boolean>> timeWindow;
+
+    public CmsswCrashes() {
+        timeWindow = new LinkedList<>();
+        this.name = "CMSSW crashes";
+
+
+        this.action = new SimpleAction("Call the HLT DOC, mentioning the messages under you see under HLT Alerts in F3 Mon. ",
+                "Call the DAQ DOC. He might need to clean up the Filter Farm.");
+
+    }
+
+    @Override
+    public boolean satisfied(DAQ daq, Map<String, Boolean> results) {
+
+        assignPriority(results);
+
+        if(daq.getHltInfo() == null){
+            return false;
+        }
+
+        int currentCrashes = daq.getHltInfo().getCrashes();
+        long startTimestampOfSlidingWindow = daq.getLastUpdate() - slidingWindowPeriodInSeconds * 1000;
+        long holdoffLimit = daq.getLastUpdate() - holdoffPeriodInSeconds * 1000;
+
+        Triple<Long, Integer, Boolean> reference = timeWindow.stream().filter(e -> e.getLeft() >= startTimestampOfSlidingWindow).findFirst().orElse(null);
+
+        boolean result = false;
+        if (reference != null) {
+
+            int currentCrashIncrement = currentCrashes - reference.getMiddle();
+            long currentTimeIncrement = daq.getLastUpdate() - reference.getLeft();
+
+            logger.debug("Current time window: " + timeWindow + " reference element: " + reference + ", current increment: " + currentCrashIncrement);
+
+            if (crashesCountThreshold <= currentCrashIncrement) {
+                float crashesPerSecond = currentCrashIncrement / (0.001f*currentTimeIncrement);
+                context.registerForStatistics("CRASHES", crashesPerSecond, " crashes/s", 1);
+                result = true;
+            }
+
+        }
+
+        timeWindow.add(Triple.of(daq.getLastUpdate(), currentCrashes, result));
+
+        // update the sliding window - keep as less data as possible for next iteration
+        timeWindow = timeWindow.stream().filter(e -> e.getLeft() >= Math.min(startTimestampOfSlidingWindow, holdoffLimit)).collect(Collectors.toList());
+
+        if (!result) {
+            logger.debug("Filtered: " + timeWindow.stream().filter(e -> e.getLeft() > holdoffLimit).collect(Collectors.toList()));
+            return timeWindow.stream().filter(e -> e.getLeft() > holdoffLimit).anyMatch(e -> e.getRight());
+        } else
+            return true;
+
+
+    }
+
+    @Override
+    public void parametrize(Properties properties) {
+        this.crashesCountThreshold = FailFastParameterReader.getIntegerParameter(properties, Setting.EXPERT_CMSSW_CRASHES_THRESHOLD, this.getClass());
+        this.slidingWindowPeriodInSeconds = FailFastParameterReader.getIntegerParameter(properties, Setting.EXPERT_CMSSW_CRASHES_TIME_WINDOW, this.getClass());
+        this.holdoffPeriodInSeconds = FailFastParameterReader.getIntegerParameter(properties, Setting.EXPERT_CMSSW_CRASHES_HOLDOFF, this.getClass());
```

Comment:
previously time parameters (e.g. `expert.logic.continoussofterror.threshold.period`) were specified in milliseconds while these seem to be in seconds. Should we globally move to seconds instead (this is more human friendly in the configuration files) ?

Reply:
That's valid point, the seconds may be more readable. Let's convert everything to seconds after integrating all of the deadtime analalysis related branches. We could actually also review the properties names convention.

Reply:
ok, opened #161 to keep track of this for the other modules which use milliseconds.